### ISSUE TITLE
[no-undef] doesn't work for return types

### ISSUE THREAD

**Issue by mjackson** [typescript-eslint/typescript-eslint#1904]
I can't get `no-undef` to warn me about missing types. I didn't see a `no-undef` rule in [the `@typescript-eslint` rules](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/docs/rules), so I'd assume I could use the same `no-undef` rule that is present in [core ESLint](https://eslint.org/docs/rules/no-undef).

**Repro**

```JSON
{
  "rules": {
    "no-undef": "error"
  }
}
```

```TS
export function fun(): ReturnType {
  console.log('wooo!');
}
```

**Expected Result**

A warning about `ReturnType` being undefined.

**Actual Result**

Nothing.

**Additional Info**

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.19.0` |
| `@typescript-eslint/parser`        | `2.19.0` |
| `TypeScript`                       | `3.8.3` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `12.13.0` |
| `npm`                              | `6.14.2` |

**Comment by bradzacher** [comment]
See #1856
